Pilot/pilotCommands.py

```diff
@@ -513,23 +513,21 @@ def execute(self):
             self.log.error("The pilot command output is not json compatible.")
             sys.exit(1)
 
-        self.pp.queueParameters = resourceDict
-        for queueParamName, queueParamValue in self.pp.queueParameters.items():
-            if isinstance(queueParamValue, list):  # for the tags
-                queueParamValue = ",".join([str(qpv).strip() for qpv in queueParamValue])
-            self.cfg.append("-o /LocalSite/%s=%s" % (queueParamName, quote(queueParamValue)))
-
         # Pick up all the relevant resource parameters that will be used in the job matching
         if "WholeNode" in resourceDict:
             self.pp.tags.append("WholeNode")
 
         # Tags must be added to already defined tags if any
-        if resourceDict.get("Tag"):
-            self.pp.tags += resourceDict["Tag"]
+        self.pp.tags += resourceDict.pop("Tag", [])
 
         # RequiredTags are like Tags.
-        if resourceDict.get("RequiredTag"):
-            self.pp.reqtags += resourceDict["RequiredTag"]
+        self.pp.reqtags += resourceDict.pop("RequiredTag", [])
+
+        self.pp.queueParameters = resourceDict
+        for queueParamName, queueParamValue in self.pp.queueParameters.items():
+            if isinstance(queueParamValue, list):  # for the tags
+                queueParamValue = ",".join([str(qpv).strip() for qpv in queueParamValue])
+            self.cfg.append("-o /LocalSite/%s=%s" % (queueParamName, quote(queueParamValue)))
```

Comment:
In the description of the PR you wrote:

```
`tags/reqtags` are only present in `/Resources/Computing/CEDefaults`
```
but here you are adding them to `/LocalSite`?

Reply:
I add them in `self.pp.tags/reqtags` and pop them from `resourceDict` before `self.pp.queueParameters = resourceDict`, so they should not be present in `/LocalSite`, am I doing it wrong?